hack/test-update-storage-objects.sh

```diff
@@ -82,55 +86,85 @@ trap cleanup EXIT SIGINT
 
+  KUBE_API_VERSIONS="${KUBE_NEW_API_VERSION},${KUBE_OLD_API_VERSION}"
+  RUNTIME_CONFIG="api/all=false,api/${KUBE_OLD_API_VERSION}=true,api/${KUBE_NEW_API_VERSION}=true"
+  startApiServer ${KUBE_NEW_STORAGE_VERSIONS}
 
-#######################################################
-# Step 2: Start a server which supports both the old and new api versions,
-# but KUBE_NEW_API_VERSION is the latest (storage) version.
-#######################################################
+  # Update etcd objects, so that will now be stored in the new api version.
+  ${UPDATE_ETCD_OBJECTS_SCRIPT}
 
-KUBE_API_VERSIONS="${KUBE_NEW_API_VERSION},${KUBE_OLD_API_VERSION}"
-RUNTIME_CONFIG="api/all=false,api/${KUBE_OLD_API_VERSION}=true,api/${KUBE_NEW_API_VERSION}=true"
-startApiServer
+  # Verify that the storage version was changed in etcd
+  curl -s http://${ETCD_HOST}:${ETCD_PORT}/v2/keys/registry/${resource}/${namespace}/${name} | grep ${KUBE_NEW_STORAGE_VERSIONS}
 
-# Update etcd objects, so that will now be stored in the new api version.
-${UPDATE_ETCD_OBJECTS_SCRIPT}
+  killApiServer
 
-killApiServer
 
+  #######################################################
+  # Step 3 : Start a server which supports only the new api version.
+  #######################################################
 
+  KUBE_API_VERSIONS="${KUBE_NEW_API_VERSION}"
+  RUNTIME_CONFIG="api/all=false,api/${KUBE_NEW_API_VERSION}=true"
+
+  # This seems to reduce flakiness.
+  sleep 1
+  startApiServer ${KUBE_NEW_STORAGE_VERSIONS}
+
+  # Verify that the server is able to read the object.
+  ${KUBECTL} get "${resource}"
+
+  killApiServer
+}
 
-#######################################################
-# Step 3 : Start a server which supports only the new api version.
-#######################################################
+# No need to test pods until there's v1 and some other newer version
+# run-upgrade-test "pods" "docs/user-guide/pod.yaml"
 
-KUBE_API_VERSIONS="${KUBE_NEW_API_VERSION}"
-RUNTIME_CONFIG="api/all=false,api/${KUBE_NEW_API_VERSION}=true"
 
-# This seems to reduce flakiness.
-sleep 1
-startApiServer
+# need to include extensions/v1beta1 in new api version because its internal types are used by jobs
+KUBE_OLD_API_VERSION="v1,extensions/v1beta1" \
+  KUBE_NEW_API_VERSION="v1,extensions/v1beta1,batch/v1" \
+  KUBE_OLD_STORAGE_VERSIONS="batch=extensions/v1beta1" \
+  KUBE_NEW_STORAGE_VERSIONS="batch/v1" \
+  run-upgrade-test "default" "jobs" "pi" "docs/user-guide/job.yaml"
 
-# Verify that the server is able to read the object.
-# This will fail if the object is in a version that is not understood by the
-# master.
-${KUBECTL} get pods
+# need to include extensions/v1beta1 in new api version because its internal types are used by hpas
+KUBE_OLD_API_VERSION="v1,extensions/v1beta1" \
+  KUBE_NEW_API_VERSION="v1,extensions/v1beta1,autoscaling/v1" \
+  KUBE_OLD_STORAGE_VERSIONS="autoscaling=extensions/v1beta1" \
+  KUBE_NEW_STORAGE_VERSIONS="autoscaling/v1" \
```

Comment:
Confirming that we dont need to write this as KUBE_NEW_STORAGE_VERSIONS="autoscaling=autoscaling/v1"?